### PR TITLE
Fix #573: Ensure that store builds use OpenXR

### LINKED ISSUE #573: Ensure that store builds use OpenXR
Right now Oculus builds can still use both OculusVR or OpenXR backend. One just need to change the value of `openxr=true` to `false` in the `user.properties` file to use one or the other. As easy as it seems it's really prone error. I did once submit a non OpenXR build to an store and don't want to commit that mistake again.

One good first step would be to cause the build to fail whenever we were trying to build a store flavor with `openxr=false`

### PR DESCRIPTION
**Pull request by kavi-kumaran**
This pull request fixes #573. It prevents from submitting a non OpenXR build to an store. It throw an error if a store flavor is built with openxr=false.

### CODE REVIEW

**Comment by kavi-kumaran** [comment]
Tagging @svillar for review

File: app/build.gradle [hunk 1]
```diff
@@ -385,6 +385,11 @@ android {
         // MetaStore and AppLab only apply to oculusvr builds.
         if ((store == 'metaStore' || store == "appLab") && !platform.startsWith('oculusvr'))
             variant.setIgnore(true);
+        
+        // Throw an error if a store flavor is built with openxr=false
+        if (store == 'generic' && !project.hasProperty('openxr') || project.getProperty('openxr') == false) {
```
Comment (svillar) [inline app/build.gradle:390]: Hmm but `generic`is precisely used for non store builds. The point is that store builds must always use `openxr=true`

**Review by svillar — CHANGES_REQUESTED** [review]
Needs some changes. Also please do not include the first 3 commits (you can remove them locally and then force-push)

**Comment by svillar** [comment]
Hmm so what happened here? are you planning to work on this?

**Comment by kavi-kumaran** [comment]
> Hmm so what happened here? are you planning to work on this?

Yes

**Pull request closed without merging**